### PR TITLE
Day 6 Part 1 & 2

### PR DESCRIPTION
**Pull request by doniacld**

### CODE REVIEW

File: internal/filereader/filereader.go [hunk 1]
```diff
@@ -51,8 +51,8 @@ func ExtractInt(path string) ([]int, error) {
 	return lines, err
 }
 
-// Extract all the values into an array
-func ExtractStringsArray(path string) ([][]string, error) {
+// ExtractStrings returns all lines in an array of array of string format
+func ExtractStrings(path string) ([][]string, error) {
```
Comment (ablqk) [inline internal/filereader/filereader.go:55]: shouldn't this take a separator in parameter?

File: puzzles/2020/06/questions/questions.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+}
+
+// IncCounter increments the counter if the value is not seen
+func (y *yesesSum) IncCounter(answers []byte) {
+	for _, a := range answers {
+		if _, ok := y.answers[a]; !ok {
+			y.answers[a] = struct{}{}
+			y.yesCounter++
+		}
+	}
```
Comment (ablqk) [inline puzzles/2020/06/questions/questions.go:35]: Alternatively:
```
	for _, a := range answers {
		y.answers[a] = struct{}{}
	}
	y.yesCounter = len(y.answers) // this probably goes l.25
```

File: puzzles/2020/06/questions/questions.go [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+	for _, a := range answers {
+		count, ok := c.answers[a]
+		if !ok {
+			// first time the question is seen
+			count = 1
+		} else {
+			// increment the existing question occurrence
+			count++
+		}
+		c.answers[a] = count
```
Comment (ablqk) [inline puzzles/2020/06/questions/questions.go:80]: Note that `myMap[absentValue]` returns the zero-value. For example, if answers is empty, `c.answers['a']` is `0`.
This means you can simply `c.answer[a]++`

File: puzzles/2020/06/questions/questions.go [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+func NewCommonAnswers() Questions {
+	return &commonAnswers{
+		answers: make(map[byte]int),
+	}
+}
+
+// GetYeses returns the counter of yeses
+func (c *commonAnswers) GetYeses() int {
+	var counter int
+	for _, a := range c.answers {
```
Comment (floppyzedolfin) [inline puzzles/2020/06/questions/questions.go:60]: to improve ease of reading, use "expliciter" names: here, `a` is the number of occurrences an answer has been seen

File: puzzles/2020/06/questions/questions.go (hunk 3 above)
Comment (doniacld) [reply puzzles/2020/06/questions/questions.go:80]: I did not know or I forgot it.
Thanks!

File: puzzles/2020/06/questions/questions.go (hunk 4 above)
Comment (doniacld) [reply puzzles/2020/06/questions/questions.go:60]: Replaced by 'occAnswers'. @floppyzedolfin is it enough ?

File: puzzles/2020/06/questions/questions.go (hunk 2 above)
Comment (doniacld) [reply puzzles/2020/06/questions/questions.go:35]: Bim, done! :)

**Pull request merged**